pkg/operator/ceph/controller/spec.go

```diff
@@ -615,6 +634,56 @@ func StoredLogAndCrashVolumeMount(varLogCephDir, varLibCephCrashDir string) []v1
 	}
 }
 
+// GenerateMDSLivenessProbeExecDaemon generates a liveness probe that makes sure mds daemon is present in fs map,
+// that it can be called, and that it returns 0
+func GenerateMDSLivenessProbeExecDaemon(daemonID, filesystemName string) *v1.Probe {
+	var mdsLivenessProbeCmd = []string{
+		"timeout",
+		fmt.Sprintf("%d", mdsCmdTimeout),
+		"sh",
+		"-c",
+		fmt.Sprintf(`
+outp="$(ceph fs dump --mon-host=$(ROOK_CEPH_MON_HOST) --mon-initial-members=$(ROOK_CEPH_MON_INITIAL_MEMBERS) --format json 2>&1)"
+rc=$?
+if [ $rc -ne 0 ]; then
+	echo "ceph mds dump check failed with the following output:"
+	echo "$outp" | sed -e 's/^/> /g'
+	echo "we couldn't get the information to reliably tell if the MDS should be restarted or not"
+else
```

Comment:
Bash also benefits from the same fail-fast methods that go encourages. If we return inside the `if` block, then there is no else section needed, making that seem less complicated

```suggestion
	echo "we couldn't get the information to reliably tell if the MDS should be restarted or not"
	return 0
fi
```

Reply:
Okay